### PR TITLE
Automatically choose origin (along with anchor) to make skin editor placement easier for users

### PR DESCRIPTION
**Pull request by peppy**
Users still don't really understand anchors and origins (and I kinda don't blame them – as a concept it takes a bit of brain power to get right), so let's choose for them in most cases and take the guesswork out of it.

https://github.com/ppy/osu/assets/191335/88ff78ec-0dd3-4ccc-ae65-cd2c6545a0a8

### CODE REVIEW

**Review by frenzibyte — DISMISSED** [review]
Works well otherwise.

File: osu.Game/Overlays/SkinEditor/SkinBlueprint.cs [hunk 1]
```diff
@@ -40,7 +40,9 @@ public partial class SkinBlueprint : SelectionBlueprint<ISerialisableDrawable>
 
         public override bool Contains(Vector2 screenSpacePos) => drawableQuad.Contains(screenSpacePos);
 
-        public override Vector2 ScreenSpaceSelectionPoint => drawable.ToScreenSpace(drawable.OriginPosition);
+        public override Vector2 ScreenSpaceSelectionPoint =>
+            // Important to use a stable position (not based on origin) as origin may be automatically updated during drag operations.
+            drawable.ScreenSpaceDrawQuad.Centre;
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Overlays/SkinEditor/SkinBlueprint.cs:45]: Seems like a fine fix for now, thanks.

**Comment by bdach** [comment]
I'm not super sure how much this matters but this doesn't handle rotation/flip quite right (and can spaz out a bit due to that):

https://github.com/ppy/osu/assets/20418176/1ad92047-5330-45b8-8afa-205bce3bce21

The idea here would be that the origin should be chosen in "screen space" rather than in the drawable's local space, but I dunno how easy making that work is going to be. Maybe we can start with this and deal with that as a follow-up?

**Comment by peppy** [comment]
Mmm, I'd be inclined to say we can handle it as a follow-up, based on how seldom people are going to be flipping elements to that extent.

**Comment by bdach** [comment]
Yeah I can agree with that.

**Review by bdach — APPROVED** [review]

**Pull request merged**